pkg/summary/gvk.go

```diff
@@ -0,0 +1,115 @@
+package summary
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	conditionStatusSep = "/"
+	gvkFormat          = "%s" + conditionStatusSep + "%s" + conditionStatusSep + "%s"
+)
+
+// conditionTypeStatusJSON is a custom JSON to map a complex object into a standard JSON object. It maps Groups, Versions and Kinds to
+// Conditions Types and Status, indicating with a flag if a certain condition with specific status represents an error or not.
+// Is is expected to be something like:
+//
+//	{
+//			"gvk": 				"helm.cattle.io/v1/HelmChart",
+//			"conditionMapping": [
+//				{
+//					"type": "JobCreated"
+//					"status": "True",
+//					"error": false,
+//				},
+//				{
+//					"type": "Failed"
+//					"status": "True",
+//					"error": true,
+//				},
+//				{
+//					"type": "Failed"
+//					"status": "False",
+//					"error": false,
+//				},
+//			}
```

Comment:
I think that you could be more concise with this if you used only a single item for each type, and simply stored a list of condition status strings (if any) that are considered errors. An empty list means the type is purely informational and there are no errors.
```suggestion
//				{
//					"type": "JobCreated",
//				},
//				{
//					"type": "Failed"
//					"conditions": ["True"],
//				},
//			}
```